ui/apps/platform/src/Containers/Vulnerabilities/NodeCves/Node/useNodeVulnerabilities.ts

```diff
@@ -20,9 +20,7 @@ const nodeVulnerabilitiesQuery = gql`
 export default function useNodeVulnerabilities({
     nodeId,
     query,
-    page,
-    perPage,
-    sortOption,
+    ...pagination
```

Comment:
The spread notation here is used because a follow up PR will change the type of `sortOption` to possibly be omitted from the parameter object, which will prevent destructuring the parameter object due to a type error. By capturing it in the `...pagination` object, we can pass it without having to reference it directly.

Reply:
For my understanding, besides rest `...pagination` more flexible for **optional** properties, do we expect TypeScript if hook changed so it consisted of **unexpected** properties for `getPaginationParams(pagination)` below.

Reply:
No, additional properties added to the `useNodeVulnerabilities` object argument would not result in a TypeScript error since the `...pagination` object would still satisfy the type of `Pagination` needed by `getPaginationParams()` regardless of other data that might be attached to it. The new properties _would_ be passed silently to this helper function, and would eventually be ignored since they are unused though.

This could result in a **runtime** issue though if:
- We did not destructure the incoming object argument to `getPaginationParams`, and instead mutated it in place and returned it. This could in theory result in extra runtime properties that could eventually get passed to e.g. a server API and have an effect there.
- We used some JavaScript APIs that did introspection of the incoming object and changed behavior based on some information about the object visible at runtime. This isn't done anywhere in our codebase as far as I know, and would be pretty far away from our current coding style.